statsmodels/distributions/tests/test_discrete.py

```diff
@@ -0,0 +1,38 @@
+import os
+import numpy as np
+import statsmodels.api as sm
+from scipy.stats import poisson
+from numpy.testing import (assert_, assert_raises, assert_almost_equal,
+                           assert_equal, assert_array_equal, assert_allclose,
+                           assert_array_less)
+
+class TestGenpoisson_p(object):
+    """
+    Test Generalized Poisson Destribution
+    """
+
+    def test_pmf_p1(self):
+        poisson_pmf = poisson.pmf(1, 1)
+        genpoisson_pmf = sm.distributions.genpoisson_p.pmf(1, 1, 0, 1)
+        assert_allclose(poisson_pmf, genpoisson_pmf, rtol=1e-15)
```

Comment:
We should be able to add a test for `mean` and `var` also for p different from 0 and 1, because we have explicit formulas for it.